rust/datafusion/src/optimizer/boolean_comparison.rs

```diff
@@ -0,0 +1,270 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
I suggest calling this module 'constant_folding.rs` and the optimization `ConstantFolding` -- while this particular PR has code to fold boolean constants, the idea is much more general (e.g you might imagine folding things like `A < (5+3)` into `A < 8` and that sort of thing).